PurchasesTests/SubscriberAttributes/BackendSubscriberAttributesTests.swift

```diff
@@ -0,0 +1,293 @@
+                                             subscriberAttribute2.key: subscriberAttribute2
+                                         ],
+                                         appUserID: appUserID,
+                                         completion: { (error: Error!) in
+                                             completionCallCount += 1
+                                             receivedError = error
+                                         })
+
+        expect(self.mockHTTPClient.invokedPerformRequestCount) == 1
+        expect(completionCallCount).toEventually(equal(1))
+        expect(receivedError).toNot(beNil())
+        expect(receivedError).to(beAKindOf(Error.self))
+
+        let receivedNSError = receivedError! as NSError
+        expect(receivedNSError.code) == Purchases.ErrorCode.unknownBackendError.rawValue
+        expect(receivedNSError.successfullySynced()) == true
+        expect(receivedNSError.userInfo["successfullySynced"]).toNot(beNil())
+        expect((receivedNSError.userInfo["successfullySynced"] as! NSNumber).boolValue) == true
+    }
+
+    func testPostSubscriberAttributesNoOpIfAttributesAreEmpty() {
+        var completionCallCount = 0
+        backend.postSubscriberAttributes([:],
+                                         appUserID: appUserID,
+                                         completion: { (error: Error!) in
+                                             completionCallCount += 1
+                                         })
+        expect(self.mockHTTPClient.invokedPerformRequestCount) == 0
+    }
+
+//    func testPostSubscriberAttributesErrorIncludesRCSuccessfullySyncedKeyIfInternalServerError() {
+//        var completionCallCount = 0
+//        mockHTTPClient.shouldInvokeCompletion = true
+//        mockHTTPClient.stubbedCompletionStatusCode = 503
+//        mockHTTPClient.stubbedCompletionError = nil
+//
+//        var receivedError: Error? = nil
+//        backend.postSubscriberAttributes([
+//                                             subscriberAttribute1.key: subscriberAttribute1,
+//                                             subscriberAttribute2.key: subscriberAttribute2
+//                                         ],
+//                                         appUserID: appUserID,
+//                                         completion: { (error: Error!) in
+//                                             completionCallCount += 1
+//                                             receivedError = error
+//                                         })
+//
+//        expect(self.mockHTTPClient.invokedPerformRequestCount) == 1
+//        expect(completionCallCount).toEventually(equal(1))
+//        expect(receivedError).toNot(beNil())
+//        expect(receivedError).to(beAKindOf(Error.self))
+//
+//        let receivedNSError = receivedError! as NSError
+//        expect(receivedNSError.code) == Purchases.ErrorCode.unknownBackendError.rawValue
+//    }
```

Comment:
Why is this commented out? Is it supposed to work only in following PRs?

Reply:
woooah, forgot about it, sorry!

Reply:
it was meant for this one, forgot to uncomment

Reply:
actually, it was a copy/paste while testing stuff, but it's duplicated from an existing test. killing